openmls/src/group/core_group/create_commit.rs

```diff
@@ -38,28 +38,9 @@ impl CoreGroup {
     ) -> Result<CreateCommitResult, CreateCommitError<KeyStore::Error>> {
         let ciphersuite = self.ciphersuite();
 
-        // If this is an external commit, we don't have an `own_leaf_index` set
-        // yet. Instead, we use the index in which we will be put in course of
-        // this commit. Our index is determined as if we'd be added through an
-        // Add proposal. However, since this might be the "resync" flavour of an
-        // external commit, it could be that we're first removing our past self
-        // from the group, in which case, we can't just take the next free leaf
-        // in the existing tree. Note, that we have to determine the index here
-        // (before we actually add our own leaf), because it's needed in the
-        // process of proposal filtering and application.
-        let (sender, own_leaf_index) = match params.commit_type() {
-            CommitType::External => {
-                // If this is a "resync" external commit, it should contain a
-                // `remove` proposal with the index of our previous self in the
-                // group.
-                let leaf_index =
-                    self.free_leaf_index(params.inline_proposals().iter().map(Some))?;
-                (Sender::NewMemberCommit, leaf_index)
-            }
-            CommitType::Member => (
-                Sender::build_member(self.own_leaf_index()),
-                self.own_leaf_index(),
-            ),
```

Comment:
The leaf index calculation here didn't take the "resync" case into account. Index computation now happens one layer above, where the index is set properly in the `CoreGroup`.